### ISSUE TITLE
[Bug] JSON deserialize for xamarin forms color is not returning color value.

### ISSUE THREAD

**Issue by saiganesh-sakthivel** [xamarin/Xamarin.Forms#14932]
### Description
JSON deserialize for xamarin forms color is not returning color value.

### Steps to Reproduce

1. SerializeObject of xamarin forms color
2. and then  deserialize the string to the xamarin forms color
3. value will be zero

### Expected Behavior
It should return the actual color value.

### Actual Behavior
It does not return the correct value.

### Basic Information

- Version with issue:
- Last known good version:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library / AndroidX Version: 
- NuGet Packages:
- Affected Devices:

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
```

</details>

### Code Snippet

Meeting clientMeeting = new Meeting();
            DateTime currentDate = DateTime.Now.AddDays(1);
            DateTime startTime = new DateTime(currentDate.Year, currentDate.Month, currentDate.Day, 10, 0, 0);
            DateTime endTime = new DateTime(currentDate.Year, currentDate.Month, currentDate.Day, 12, 0, 0);
            clientMeeting.From = startTime;
            clientMeeting.To = endTime;
            clientMeeting.Color = Color.Blue;
            clientMeeting.EventName= "ClientMeeting";

            Meeting test = JsonConvert.DeserializeObject<Meeting>(JsonConvert.SerializeObject(clientMeeting));
            StartTime.Text = test.From.ToLongDateString();
            EndTime.Text = test.To.ToLongDateString();
            EventName.Text = test.EventName.ToString();
            Stack.BackgroundColor = test.Color;

Xaml
              **<ContentPage.Content>
                    <StackLayout x:Name="Stack">
                        <Label x:Name="StartTime"/>
                        <Label x:Name="EndTime"/>
                        <Label x:Name="EventName"/>
                    </StackLayout>
                </ContentPage.Content>**

### Reproduction Link

https://www.syncfusion.com/downloads/support/directtrac/general/ze/ScheduleXamarin-776150168

**Comment by jfversluis** [comment]
I don't believe this is a bug but due to a combination of `Color` being a `struct` and how that is serialized and deserialized. I have made the following changes in your code:

```csharp
var json = JsonConvert.SerializeObject(clientMeeting, new JsonSerializerSettings
{
     Converters = new JsonConverter[] { new JsonColorConverter() }
});

Meeting test = JsonConvert.DeserializeObject<Meeting>(json, new JsonSerializerSettings
{
      Converters = new JsonConverter[] { new JsonColorConverter() }
});
```

Note that in the `DeserializeObject` I am now using `json` instead of serializing the same object again as in your original code. Additionally you can see I added a custom `JsonConverter` class to handle the `Color`, see the implementation below.

```csharp
        public class JsonColorConverter : JsonConverter<Color>
        {
            public override void WriteJson(JsonWriter writer, Color value, JsonSerializer serializer)
            {
                writer.WriteValue(value.ToHex());
            }

            public override Color ReadJson(JsonReader reader, Type objectType, Color existingValue, bool hasExistingValue, JsonSerializer serializer)
            {
                string s = (string)reader.Value;

                return Color.FromHex(s);
            }
        }
```

You might want to add some error handling, but this serializes it as a hex string and also retrieves it as such. This should work.
